cpp/src/arrow/flight/sql/odbc/tests/CMakeLists.txt

```diff
@@ -31,10 +31,12 @@ set(ARROW_FLIGHT_SQL_MOCK_SERVER_SRCS
 
 add_arrow_test(connection_test
                SOURCES
-               connection_test.cc
                connection_attr_test.cc
+               statement_test.cc
+               connection_test.cc
                odbc_test_suite.cc
                odbc_test_suite.h
+               ../../../../engine/substrait/protobuf_test_util.cc
```

Comment:
Can you clarify what you get from this? Perhaps this should be in a more common location?

Reply:
Yup, added comment that this is meant to enable Protobuf cleanup. Regarding the location, yea I think it makes sense to put in a common location but should be a separate issue

Reply:
Created and mentioned[ GH-46889](https://github.com/apache/arrow/issues/46889) in comment